domain/schema/model.go

```diff
@@ -46,24 +47,96 @@ CREATE TABLE model_config (
 `)
 }
 
-func spacesSchema() schema.Patch {
+func spaceSchema() schema.Patch {
 	return schema.MakePatch(`
-CREATE TABLE provider_spaces (
-    uuid            TEXT PRIMARY KEY,
-    name            TEXT
+CREATE TABLE provider_space (
+    provider_id     TEXT PRIMARY KEY,
+    space_uuid      TEXT NOT NULL,
+    CONSTRAINT      fk_provider_space_space_uuid
+        FOREIGN KEY     (space_uuid)
+        REFERENCES      space(uuid)
 );
 
-CREATE TABLE spaces (
+CREATE UNIQUE INDEX idx_provider_space_space_uuid
+ON provider_space (space_uuid);
+
+CREATE TABLE space (
     uuid            TEXT PRIMARY KEY,
     name            TEXT NOT NULL,
-    is_public       BOOLEAN,
-    provider_uuid   TEXT,
-    CONSTRAINT      fk_lease_pin_lease
-        FOREIGN KEY     (provider_uuid)
-        REFERENCES      provider_spaces(uuid)
+    is_public       BOOLEAN
 );
 
 CREATE UNIQUE INDEX idx_spaces_uuid_name
-ON spaces (uuid, name);
+ON space (name);
+`)
+}
+
+func subnetSchema() schema.Patch {
+	return schema.MakePatch(`
+CREATE TABLE provider_subnet (
+    provider_id     TEXT PRIMARY KEY,
+    subnet_uuid     TEXT NOT NULL,
+    CONSTRAINT      fk_provider_subnet_subnet_uuid
+        FOREIGN KEY     (subnet_uuid)
+        REFERENCES      subnet(uuid)
+);
+
+CREATE UNIQUE INDEX idx_provider_subnet_subnet_uuid
+ON provider_subnet (subnet_uuid);
+
+CREATE TABLE provider_network (
+    provider_network_id TEXT PRIMARY KEY,
```

Comment:
This is not right, because it would imply that a provider network contains at most one subnet.

Juju does not currently have an analogue for a _provider's_ networks, but we might as well model it correctly. This table should be uuid and provider_network_id, with a join table to provider_subnet.

Reply:
oh yes you are right, fixed. I added a new table 
```
CREATE TABLE provider_network (
    uuid                TEXT PRIMARY KEY,
    provider_network_id TEXT
);
```
in case we might have more to add on the provider network, although today it seems overkill.